hermes/views.py

```diff
@@ -173,11 +186,28 @@ def submit_to_hop(request, message):
         # open for write ('w') returns a hop.io.Producer instance
         with stream.open(f'kafka://kafka.scimma.org/{topic}', 'w') as producer:
             metadata = {'topic': topic}
-            producer.write(message, metadata)
+            payload, headers = producer.pack(message, metadata)
+            message_uuid_tuple = next((item for item in headers if item[0] == '_id'), None)
+            message_uuid = None
+            if message_uuid_tuple:
+                message_uuid = uuid.UUID(bytes=message_uuid_tuple[1])
+            producer.write_raw(payload, headers)
     except Exception as e:
-        return Response({'message': f'Error posting message to kafka: {e}'},
-                        status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-    return Response({"message": "Message was submitted successfully."}, status=status.HTTP_200_OK)
+        raise APIException(f'Error posting message to kafka: {e}')
+
+    return message_uuid
+
+
+def submit_to_gcn(request, message, message_uuid):
+    # TODO: Add code to submit the message with its message_uuid to gcn here.
+    # First add the uuid into the message, then transfer the message into plaintext format
+    message_plaintext = convert_to_plaintext(message)
+    message_plaintext += '\n\n This message can be viewed at https://hermes.lco.global/messages/' + str(message_uuid)
+    # Then submit the plaintext message to gcn via email
+    send_email(settings.GCN_EMAIL, settings.HERMES_EMAIL, settings.HERMES_EMAIL_PASSWORD,
```

Comment:
I know the email have to be secret, but for consistency's sake I have `HERMES_EMAIL_USERNAME` and `HEREMS_EMAIL_PASSWORD` defined as k8s secrets. I think keeping these values together and handling them consistently reduces cognitive complexity.